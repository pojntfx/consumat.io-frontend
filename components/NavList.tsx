import NavListItem from "./NavListItem";
import { NavigationLink } from "./Navbar";
import styles from "../styles/Navbar.module.css";

type NavListProps = {
  activeNavigationLink: NavigationLink;
  isMobile: boolean;
  isMobileNavVisible: boolean;
  setMobileNavbarVisibility?: (boolean) => void;
};

const NavList = ({
  activeNavigationLink,
  isMobile,
  isMobileNavVisible,
  setMobileNavbarVisibility = null,
}: NavListProps) => {
  return (
    <ul
      className={
        isMobile && isMobileNavVisible
          ? styles.navigationListMobile + " flex"
          : isMobile && !isMobileNavVisible
          ? styles.navigationListMobile + " hidden"
          : styles.navigationList
      }
    >
      <NavListItem
        title="Home"
        href="/"
        isMobile={isMobile}
        isActive={activeNavigationLink === NavigationLink.Home}
        setMobileNavbarVisibility={
          setMobileNavbarVisibility && setMobileNavbarVisibility
        }
      />
      <NavListItem
        title="Discover"
        href="/discover"
        isMobile={isMobile}
        isActive={activeNavigationLink === NavigationLink.Discover}
        setMobileNavbarVisibility={
          setMobileNavbarVisibility && setMobileNavbarVisibility
        }
      />
      <NavListItem
        title="Library"
        href="/library"
        isMobile={isMobile}
        isActive={activeNavigationLink === NavigationLink.Library}
        setMobileNavbarVisibility={
          setMobileNavbarVisibility && setMobileNavbarVisibility
        }
      />
      <NavListItem
        title="Search"
        href="/search"
        isMobile={isMobile}
        isActive={activeNavigationLink === NavigationLink.Search}
        setMobileNavbarVisibility={
          setMobileNavbarVisibility && setMobileNavbarVisibility
        }
      />
    </ul>
  );
};

export default NavList;
