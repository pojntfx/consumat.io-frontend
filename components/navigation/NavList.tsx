import { MoonIcon, SunIcon } from "@heroicons/react/outline";
import { useSession } from "next-auth/client";
import { useContext } from "react";
import { ThemeContext } from "../../pages/_app";
import styles from "../../styles/Navbar.module.css";
import { theme } from "../../types/theme";
import { NavigationLink } from "./Navbar";
import NavListItem from "./NavListItem";

type NavListProps = {
  activeNavigationLink: NavigationLink;
  isMobile: boolean;
  isMobileNavVisible: boolean;
  setMobileNavbarVisibility?: (boolean) => void;
};

const NavList = ({
  activeNavigationLink,
  isMobile,
  isMobileNavVisible,
  setMobileNavbarVisibility = null,
}: NavListProps) => {
  const { currentTheme, setCurrentTheme } = useContext(ThemeContext);

  const [session] = useSession();

  return (
    <ul
      className={
        isMobile && isMobileNavVisible
          ? styles.navigationListMobile + " flex"
          : isMobile && !isMobileNavVisible
          ? styles.navigationListMobile + " hidden"
          : styles.navigationList
      }
    >
      <NavListItem
        title="Home"
        href="/"
        isMobile={isMobile}
        isActive={activeNavigationLink === NavigationLink.Home}
        setMobileNavbarVisibility={
          setMobileNavbarVisibility && setMobileNavbarVisibility
        }
      />
      <NavListItem
        title="Discover"
        href="/discover"
        isMobile={isMobile}
        isActive={activeNavigationLink === NavigationLink.Discover}
        setMobileNavbarVisibility={
          setMobileNavbarVisibility && setMobileNavbarVisibility
        }
      />
      <NavListItem
        title="Library"
        href="/library"
        isMobile={isMobile}
        isActive={activeNavigationLink === NavigationLink.Library}
        setMobileNavbarVisibility={
          setMobileNavbarVisibility && setMobileNavbarVisibility
        }
      />
      <NavListItem
        title="Search"
        href="/search"
        isMobile={isMobile}
        isActive={activeNavigationLink === NavigationLink.Search}
        setMobileNavbarVisibility={
          setMobileNavbarVisibility && setMobileNavbarVisibility
        }
      />
      {session && (
        <NavListItem
          title="Account"
          href="/account"
          isMobile={isMobile}
          isActive={activeNavigationLink === NavigationLink.Account}
          setMobileNavbarVisibility={
            setMobileNavbarVisibility && setMobileNavbarVisibility
          }
        />
      )}

      {currentTheme === theme.light ? (
        <button
          className="bg-transparent text-gray-400 hover:text-gray-800 dark:hover:text-white shadow-none hover:shadow-none my-4 md:my-0 ml-1"
          onClick={() => setCurrentTheme(theme.dark)}
        >
          <MoonIcon className="h-8 w-8" />
        </button>
      ) : (
        <button
          className="bg-transparent text-gray-400 hover:text-gray-800 dark:hover:text-white shadow-none hover:shadow-none my-4 md:my-0 ml-1"
          onClick={() => setCurrentTheme(theme.light)}
        >
          <SunIcon className="h-8 w-8" />
        </button>
      )}
    </ul>
  );
};

export default NavList;
