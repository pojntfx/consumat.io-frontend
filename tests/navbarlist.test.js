import React from "react";
import renderer from "react-test-renderer";
import NavList from "../components/navigation/NavList";

it("renders correctly", () => {
  const tree = renderer.create(<NavList />).toJSON();
  expect(tree).toMatchSnapshot();
});
