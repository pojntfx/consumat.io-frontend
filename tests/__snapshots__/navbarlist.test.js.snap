// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="navigationList"
>
  <li>
    <a
      className="navigation-link link navigationLink dark:hover:border-white dark:hover:text-white false"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Home
    </a>
  </li>
  <li>
    <a
      className="navigation-link link navigationLink dark:hover:border-white dark:hover:text-white false"
      href="/library?media=TV&watchStatus=Watching"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Library
    </a>
  </li>
  <li>
    <a
      className="navigation-link link navigationLink dark:hover:border-white dark:hover:text-white false"
      href="/search?q="
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Search
    </a>
  </li>
  <button
    className="bg-transparent text-gray-400 hover:text-gray-800 dark:hover:text-white shadow-none hover:shadow-none my-4 md:my-0 ml-1"
    onClick={[Function]}
  >
    <svg
      className="h-8 w-8"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
      />
    </svg>
  </button>
</ul>
`;
